geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/data/KafkaCacheLoader.scala

```diff
@@ -89,7 +89,24 @@ trait KafkaCacheLoader extends Closeable with LazyLogging {
   }
 }
 
-object KafkaCacheLoader {
+object KafkaCacheLoader extends LazyLogging {
+  object LoaderStatus {
+    private val count = new AtomicInteger(0)
+    private val firstLoadStarTime = new AtomicLong(0L)
+
+    def startLoad(): Boolean = {
+      count.incrementAndGet()
+      firstLoadStarTime.compareAndSet(0L, System.currentTimeMillis())
+    }
+    def completedLoad(): Unit = {
+      if (count.decrementAndGet() == 0) {
+        logger.info(s"Last active initial load completed.  " +
```

Comment:
could we reset firstLoadStartTime here, so that if a new data store is loaded it will have an accurate time?

Reply:
Good call.  I like it.

Reply:
if so, there's some potential concurrency issues since we're changing the count and time separately - might be easier just to make both methods synchronized

Reply:
Do we need to synchronize the methods or the entire object?

Reply:
i don't think you can synchronize a whole object - you can just put `def foo(): Unit = sycnhronized {}` which will use a lock on the object so synchronize across all methods that are declared that way

Reply:
there is probably a more elegant way but just brute synchronization should be good enough here...

Reply:
Got it.  I thought that synchronized only locked that method.  That makes sense.